common/djangoapps/student/models.py

```diff
@@ -1602,6 +1602,11 @@ def enforce_single_login(sender, request, user, signal, **kwargs):    # pylint:
         else:
             key = None
         if user:
+            if not UserProfile.objects.filter(user=user):
```

Comment:
@justinabrahms this is user profile
- What if user change his name,  then next time he login  ``name not equal user.username`` the it will create user profile again.
- This code trigger each time user login
- for the second and onward this [code](https://github.com/edx/edx-platform/blob/master/common/djangoapps/external_auth/views.py#L481) fixed profile issue